lib/strategy/FragmentationStrategyShape.ts

```diff
@@ -0,0 +1,197 @@
+import { readFileSync } from 'fs';
+import { readFile } from 'fs/promises';
+import { join } from 'path';
+import type * as RDF from '@rdfjs/types';
+import * as ShexParser from '@shexjs/parser';
+import { JsonLdParser } from 'jsonld-streaming-parser';
+import { DataFactory } from 'rdf-data-factory';
+import type { IQuadSink } from '../io/IQuadSink';
+import { FragmentationStrategyStreamAdapter } from './FragmentationStrategyStreamAdapter';
+import { FragmentationStrategySubject } from './FragmentationStrategySubject';
+// eslint-disable-next-line import/extensions
+import * as SHEX_CONTEXT from './shex_context.json';
+
+const DF = new DataFactory<RDF.Quad>();
+
+interface IShapeEntry {
+  shape: string;
+  folder: string;
+}
+
+export class FragmentationStrategyShape extends FragmentationStrategyStreamAdapter {
```

Comment:
Could you add some tsdoc to the different methods and inline? Because it's hard to understand the code right now :/